core/src/main/java/org/ldaptive/intermediate/SyncInfoMessage.java

```diff
@@ -230,10 +230,38 @@ public void decode(final byte[] berValue)
 
 
   @Override
+  public boolean equals(final Object o)
+  {
+    if (o == this) {
+      return true;
+    }
+    if (o instanceof SyncInfoMessage) {
+      final SyncInfoMessage v = (SyncInfoMessage) o;
+      return LdapUtils.areEqual(getOID(), v.getOID()) &&
+             LdapUtils.areEqual(messageType, v.messageType) &&
+             LdapUtils.areEqual(cookie, v.cookie) &&
+             LdapUtils.areEqual(refreshDone, v.refreshDone) &&
+             LdapUtils.areEqual(refreshDeletes, v.refreshDeletes) &&
+             LdapUtils.areEqual(entryUuids, v.entryUuids);
```

Comment:
The fields here are not the same as hashCode -- missing controls and messageId. Probably should be same.